### PR TITLE
[Bugfix] Include entry-point logits processor plugins in output token…

### PR DESCRIPTION
**Pull request by YingxuH**
## Purpose

Fix entry-point logits processor plugins (`vllm.logits_processors` group) being silently ignored when computing the `logitsprocs_need_output_token_ids` flag in `gpu_model_runner.py`.

Previously, the flag was derived solely from `bool(custom_logitsprocs)`, which only reflects CLI-passed processors. Entry-point plugins loaded by `build_logitsprocs()` → `_load_logitsprocs_plugins()` were not accounted for. When the flag was `False` and all penalties were neutral (`repetition_penalty=1.0`), vLLM's async scheduling path filled the output token ID buffer with `-1` placeholders instead of real tokens, silently breaking any entry-point logits processor that inspects generation history (e.g., n-gram repetition prevention).

**Changes:**
- **`state.py`**: Add `has_custom` keyword-only parameter to `LogitsProcessors.__init__` (defaults to `False`, backward-compatible).
- **`__init__.py`**: `build_logitsprocs()` sets `has_custom=bool(custom_logitsprocs_classes)`, where `custom_logitsprocs_classes` includes both entry-point plugins and CLI-passed processors.
- **`gpu_model_runner.py`**: Extract `build_logitsprocs()` result into a local variable and derive the flag from `logitsprocs.has_custom or self.vllm_config.reasoning_config is not None`.

## Test Plan

New test file `tests/v1/logits_processors/test_entrypoint_output_token_tracking.py` with 3 test cases, reusing existing mock utilities from `tests/v1/logits_processors/utils.py`:

```bash
# New tests
python -m pytest tests/v1/logits_processors/test_entrypoint_output_token_tracking.py -v

# Existing correctness tests (regression check)
python -m pytest tests/v1/logits_processors/test_correctness.py -v
```

## Test Result

Built from source on H100 (CUDA 12.8, Python 3.12, PyTorch 2.10):

```
tests/v1/logits_processors/test_entrypoint_output_token_tracking.py::test_entrypoint_plugin_enables_output_token_tracking PASSED
tests/v1/logits_processors/test_entrypoint_output_token_tracking.py::test_cli_logitsprocs_still_enable_tracking PASSED
tests/v1/logits_processors/test_entrypoint_output_token_tracking.py::test_no_logitsprocs_disables_tracking PASSED

tests/v1/logits_processors/test_correctness.py::test_logitsprocs[logitsprocs_under_test0-50-cuda:0] PASSED
tests/v1/logits_processors/test_correctness.py::test_logitsprocs[logitsprocs_under_test1-50-cuda:0] PASSED
tests/v1/logits_processors/test_correctness.py::test_logitsprocs[logitsprocs_under_test2-50-cuda:0] PASSED
tests/v1/logits_processors/test_correctness.py::test_logitsprocs[logitsprocs_under_test3-50-cuda:0] PASSED
tests/v1/logits_processors/test_correctness.py::test_logitsprocs[logitsprocs_under_test4-50-cuda:0] PASSED
tests/v1/logits_processors/test_correctness.py::test_logitsprocs[logitsprocs_under_test5-50-cuda:0] PASSED
```

---

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update.

### CODE REVIEW

File: vllm/v1/worker/gpu_model_runner.py [hunk 1]
```diff
@@ -614,6 +614,25 @@ def __init__(
...
+            self.pin_memory,
+            self.is_pooling_model,
+            custom_logitsprocs,
+        )
+        # Check whether any custom logits processor (CLI-passed OR
+        # entry-point plugin) is loaded.  Previously this flag was derived
+        # from `custom_logitsprocs` alone, which only contains CLI-passed
+        # processors.  Entry-point plugins loaded by build_logitsprocs()
+        # were not accounted for, causing their output_token_ids buffers to
+        # be filled with -1 placeholders when all penalties are neutral.
```
Comment (njhill) [inline vllm/v1/worker/gpu_model_runner.py]: I think these comment lines can be removed, we don't want to document history in the code.

**Review by njhill — COMMENTED** [review]
Thanks @YingxuH. LGTM, just one comment.

File: vllm/v1/worker/gpu_model_runner.py (hunk 1 above)
Comment (YingxuH) [reply vllm/v1/worker/gpu_model_runner.py]: Thanks for your review. I have removed the history lines and rebased onto the latest main branch.

**Comment by YingxuH** [comment]
@njhill Addressed your feedback 8 days ago — is there anything else needed from my side?
